### ISSUE TITLE
Inconsistency in found number of topics

### ISSUE THREAD

**Issue by bmanczak** [ddangelov/Top2Vec#140]
Hi,

I wanted to try out the library with the tutorial on the 20NewsGroup dataset.

I've changed one small thing, namely only used the train data. Surprisingly, the model only found 3 topics, whereas the example in the README trained on the whole dataset found 77.
So I've copy pasted the model training code snippet and the resulting model found 114 topics.
Please find the results of the experiment in the image below:

<img width="823" alt="Screenshot 2021-02-26 at 12 37 33" src="https://user-images.githubusercontent.com/20094977/109295667-6a785880-782f-11eb-85a1-8e236fe74a29.png">

I can't figure out where these drastic differences are coming from? I understand there is stochasticity in training some models but such wild differences in results are worrying and might point at a possible bug.

**Comment by ddangelov** [comment]
The example in the readme used the `speed='fast-learn'`, in this case `doc2vec` uses negative sampling instead of a hierarchical softmax, which leads to it learning different document vectors. Using `speed='learn'` will lead to a greater number of topics since the doc2vec model uses hierarchical softmax which will lead to better document vectors. 

The 20 newsgroups test data contains only around 5000 samples, where as the train has around 13000. In the example both were used. Using a small dataset will not allow doc2vec to learn very good document and word vectors which will lead to few topics being found. This is to be expected since good word and document vectors are required. For smaller datasets I would recommend using `embedding_model=universal-sentence-encoder-multilingual`.

**Comment by bmanczak** [comment]
Okay, good point about `doc2vec` performing poorly with a small support set.

I tried running Top2Vec with `subset = "train",  embedding_model = "universal-sentence-encoder"` corpus of 20NewsGroups and only 3 topics were detected. With `subset = "all",  embedding_model = "universal-sentence-encoder"` 103 topic were detected. 

It is not to be expected behaviour, is it? What hyper parameters would you change to mitigate that?

**Comment by MattiaPance** [comment]
@ddangelov I think you should consider re-opening this issue.

I ran ten times top2Vec on the first 15K documents of the 20Newsgropus dataset with `embedding_model = "universal-sentence-encoder"` and the same set of parameters. I obtained each time a different number of topics (from 2 to 104).

**Comment by ddangelov** [comment]
Could you all specify what versions of Top2Vec, UMAP and HSBSCAN you are using?

**Comment by MattiaPance** [comment]
top2vec==1.0.22
umap-learn==0.5.1
hdbscan==0.8.27

**Comment by bmanczak** [comment]
top2vec==1.0.23
umap-learn==0.5.1
hdbscan==0.8.27
